### PR TITLE
Fix #74. Add method section

### LINKED ISSUE #74: Add method section
Add sections about
- CI/CD
- Docker
- Container
- Next.js
- Typescript
- Firebase
- Stockfish
- Python
- Images

### PR DESCRIPTION
**Pull request by Danar435**

### CODE REVIEW

File: src/method/technology.tex [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+learned the basics of the language and were able to use it effectively.
+
+The frontend of the project was built with React~\cite{react}, which is a JavaScript / TypeScript library for building
+user interfaces.
+It is developed and maintained by Meta, and is used by many large companies.
+React is focused on mobile applications first and foremost, but the team's focus was on the desktop experience.
+
+% textidote: ignore begin % due to node.js and next.js being identified as an url
+The backend was built with Next.js~\cite{next.js}, which is a React framework that allows for server-side rendering.
+The team chose Next.js over other frameworks like Node.js~\cite{node.js}, because it is newer and more modern.
```
Comment (SebastianMygind) [inline src/method/technology.tex]: This is vague in context to the relationship between node and next. Next.js does not substitute node, but runs on top of node.

File: src/method/technology.tex [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+Next.js also integrates well with React, which made it easier to build the frontend and backend together.
+% textidote: ignore end
+
+For interacting with Stockfish~\cite{stockfish}, the team created an API using Python.
+They had to use Python, as it's the only language that Stockfish supports.
```
Comment (SebastianMygind) [inline src/method/technology.tex]: It was not so much that Python was the only choice, but that it was the simplest choice for quick development. Stockfish is a C++ CLI program, and needs to be parsed to use in our program. The python stockfish parser was chosen for simplicity in regards to parsing and making an simple API.

**Review by SebastianMygind — CHANGES_REQUESTED** [review]
See comments.

**Review by SebastianMygind — DISMISSED** [review]
Nice.

File: src/method/technology.tex [hunk 3]
```diff
@@ -0,0 +1,100 @@
+\section{Technology}\label{sec:technology}
+
+Throughout the development of the project, a number of different tools and publicly available packages were used to ease
```
Comment (MadsHeilmann) [inline src/method/technology.tex]: Maybe say assist instead of ease.

File: src/method/technology.tex [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+the development process.
+These different technologies were chosen based on the team's needs and the requirements of the project.
+The following sections will describe the different technologies used in the project.
+
+\subsection{Programming stack}\label{subsec:programming}
+
+As the chess game would be hosted on a website, and will require multiple users to interact with each other, the team
```
Comment (MadsHeilmann) [inline src/method/technology.tex]: Stay in present tense and change "would" to "will".

Comment (MadsHeilmann) [inline src/method/technology.tex]: Not a good argument for why we use TypeScript. I think it is too indirect. Maybe just say: "As the project is a web application, TypeScript is used".

File: src/method/technology.tex [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+decided to use TypeScript~\cite{typescript} as the main programming language.
+TypeScript is a superset of JavaScript, with the main difference being that TypeScript is statically typed, while
+JavaScript is dynamically typed.
+Typescript can be very useful in larger projects, as it can help catch errors before the code is run.
```
Comment (MadsHeilmann) [inline src/method/technology.tex]: Maybe use more technical terms like "during compile-time" and "during run-time"

File: src/method/technology.tex [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+While many in the team did not have any prior experience with TypeScript, in comparison to JavaScript, they quickly
+learned the basics of the language and were able to use it effectively.
```
Comment (MadsHeilmann) [inline src/method/technology.tex]: Becomes a little too storytelling-ish in my opinion.

File: src/method/technology.tex [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+The frontend of the project was built with React~\cite{react}, which is a JavaScript / TypeScript library for building
+user interfaces.
+It is developed and maintained by Meta, and is used by many large companies.
+React is focused on mobile applications first and foremost, but the team's focus was on the desktop experience.
+
+% textidote: ignore begin % due to node.js and next.js being identified as an url
+The backend was built using Node.js~\cite{node.js}, which allows the team to build the server using JavaScript.
```
Comment (MadsHeilmann) [inline src/method/technology.tex]: A little too inaccurate in my opinion, as we are not building servers. Mayby instead say "which allows the team to build server-side components using TypeScript".

File: src/method/technology.tex [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+Node.js is a popular choice for backend development, as it is fast and scalable.
+As the team is using React for client-side rendering, they decided to also incorporate Next.js~\cite{next.js} to handle
+the server-side rendering.
```
Comment (MadsHeilmann) [inline src/method/technology.tex]: Again a little too storytelling-ish. Could instead be: "As well as using React for client-side rendering, Next.js is incorporated to handle the server-side rendering".

File: src/method/technology.tex [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+Next.js is a React framework running on top of Node.js, which makes it easier to build the frontend and backend
+together.
```
Comment (MadsHeilmann) [inline src/method/technology.tex]: Maybe change "which makes it easier" to "which assists in building..."

File: src/method/technology.tex [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+% textidote: ignore end
+
+As Stockfish~\cite{stockfish} is built on C++, the team had to find a way to interact with it.
+To do this, the team created a REST API using Python, as it was deemed the easiest to implement using
+py-stockfish~\cite{py-stockfish} as a wrapper.
```
Comment (MadsHeilmann) [inline src/method/technology.tex]: Again a little too storytelling-esque in my opinion. Also remember to use present tense.

File: src/method/technology.tex [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+The API was used to create functions that would take input from the server, run it through Stockfish, and return the
+output back to the server.
+Unit tests were used to ensure that the API was working correctly.
```
Comment (MadsHeilmann) [inline src/method/technology.tex]: Use present tense.

File: src/method/technology.tex [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+other.
+
+\subsection{Hosting provider}\label{subsec:hosting}
+
+When it came to hosting the website, the team had to choose between a number of different providers.
+Amazon Web Services~\cite{aws} was considered, but it was deemed too expensive for the team's needs.
+Instead, the team settled down on Firebase~\cite{firebase}, which is a platform developed by Google.
+The main reason behind this decision was that Firebase allowed for easy authentication with Google accounts, which made
+it easier to integrate user accounts into the website.
+Firebase is also free up to a certain point, which suited the team's budget.
```
Comment (MadsHeilmann) [inline src/method/technology.tex]: It sounds a little like a novel. In my opinion it reads a little too much like storytelling. Consider changing it to a more straightforward tone.

**Review by MadsHeilmann — CHANGES_REQUESTED** [review]
Overall i believe you are spitting straight facts in your writing of this segment. Big up for that. However, I think too much of it reads too much like storytelling. Some of the phrasing is unecessary and does not add value to the good info you are providing. An example could be this sentence:

"The team considered using Socket.io~\cite{socket.io} for real-time communication, but by picking Firebase, they
couldn't use it anymore."

That sentence has a little too much yapping.

The team may not agree, but in my opinion changing it to the below sentence is much smoother and reads more academic:

"For real-time communication Firebase is used. Another tool that is commonly used for similar purposes is Socket.io~\cite{socket.io}. However, due to XX reason it is not utilized."

This is a little more direct and has none of the storytelling elements.

Another point is to stay in present tense as we are doing throughout the report.

Let me know if you need clarifications on anything i wrote in this review :)

**Comment by Danar435** [comment]
Thanks for the feedback. You're right about it being more like a story, because honestly, I'm better at writing stories than reports! I rewrote most paragraphs, so it should be better now. Also I didn't realize that it had to be present tense, my bad on that one.

**Review by MadsHeilmann — DISMISSED** [review]
LGTM! :)

File: .github/workflows/lint-and-compile.yml [hunk 13]
```diff
@@ -26,6 +26,7 @@ jobs:
         uses: xu-cheng/latex-action@v3
         with:
           root_file: main.tex
+          docker_image: ghcr.io/xu-cheng/texlive-full:20240315
```
Comment (audio-engineer) [inline .github/workflows/lint-and-compile.yml]: Remove this again

**Review by audio-engineer — CHANGES_REQUESTED** [review]

**Review by MadsHeilmann — DISMISSED** [review]

File: src/method/technology.tex [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+It is developed and maintained by Meta, and is used by many large companies.
+React is focused on mobile applications first and foremost, but the team's focus was on the desktop experience.
+
+% textidote: ignore begin % due to node.js and next.js being identified as an url
+The backend is built using Node.js~\cite{node.js}, which allows the team to build server-side components using
+TypeScript.
+Node.js is a popular choice for backend development, as it is fast and scalable.
+As well as using React for client-side rendering, Next.js~\cite{next.js} is incorporated to handle the server-side
+rendering.
+Next.js is a React framework running on top of Node.js, which which assists in building the frontend and backend
```
Comment (TexWO) [inline src/method/technology.tex]: 'Which' two times line 28

Comment (Danar435) [reply src/method/technology.tex]: I blame this on textidote (which I ignore in this case)

**Review by TexWO — APPROVED** [review]

**Review by audio-engineer — APPROVED** [review]

**Pull request merged**